File: android-app-v2/app/src/main/kotlin/com/mediaplayerapi/model/ApiResponse.kt
package com.mediaplayerapi.model

import com.google.gson.annotations.SerializedName
import java.text.SimpleDateFormat
import java.util.*

/**
 * API Response wrappers matching the Python version's format.
 */
data class ApiResponse(
    @SerializedName("system")
    val system: SystemInfo,

    @SerializedName("media")
    val media: MediaInfo?,

    @SerializedName("last_updated")
    val lastUpdated: String,

    @SerializedName("cached")
    val cached: Boolean = false,

    @SerializedName("error")
    val error: String? = null
) {
    companion object {
        private val dateFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSXXX", Locale.US)

        fun now(): String = dateFormat.format(Date())
    }
}

/**
 * Status response for root endpoint.
 */
data class StatusResponse(
    @SerializedName("status")
    val status: String = "running",

    @SerializedName("service")
    val service: String = "Media Player API",

    @SerializedName("version")
    val version: String = "2.0.0",

    @SerializedName("system")
    val system: SystemInfo,

    @SerializedName("provider")
    val provider: String = "MediaSession",

    @SerializedName("cache_ttl_seconds")
    val cacheTtlSeconds: Int = 30
)

File: desktop-app/src/DesktopApp/Models/ApiResponse.cs
using System.Text.Json.Serialization;

namespace DesktopApp.Models;

/// <summary>
/// System information returned by the API.
/// </summary>
public record SystemInfo
{
    [JsonPropertyName("os")]
    public string Os { get; init; } = string.Empty;

    [JsonPropertyName("hostname")]
    public string Hostname { get; init; } = string.Empty;

    [JsonPropertyName("platform")]
    public string Platform { get; init; } = string.Empty;
}

/// <summary>
/// Full response from /now-playing endpoint.
/// </summary>
public record ApiResponse
{
    [JsonPropertyName("system")]
    public SystemInfo? System { get; init; }

    [JsonPropertyName("media")]
    public MediaInfo? Media { get; init; }

    [JsonPropertyName("last_updated")]
    public string LastUpdated { get; init; } = string.Empty;

    [JsonPropertyName("cached")]
    public bool Cached { get; init; }

    [JsonPropertyName("error")]
    public string? Error { get; init; }
}

/// <summary>
/// Response from the root / endpoint.
/// </summary>
public record StatusResponse
{
    [JsonPropertyName("status")]
    public string Status { get; init; } = string.Empty;

    [JsonPropertyName("service")]
    public string Service { get; init; } = string.Empty;

    [JsonPropertyName("version")]
    public string Version { get; init; } = string.Empty;

    [JsonPropertyName("system")]
    public SystemInfo? System { get; init; }

    [JsonPropertyName("provider")]
    public string? Provider { get; init; }

    [JsonPropertyName("cache_ttl_seconds")]
    public int CacheTtlSeconds { get; init; } = 30;
}

/// <summary>
/// WebSocket message from /ws endpoint.
/// </summary>
public record WsMessage
{
    [JsonPropertyName("type")]
    public string Type { get; init; } = string.Empty;

    [JsonPropertyName("system")]
    public SystemInfo? System { get; init; }

    [JsonPropertyName("media")]
    public MediaInfo? Media { get; init; }

    [JsonPropertyName("last_updated")]
    public string LastUpdated { get; init; } = string.Empty;

    [JsonPropertyName("cached")]
    public bool Cached { get; init; }
}
